pkg/controller/sync/controller.go

```diff
@@ -759,15 +770,24 @@ func (s *FederationSyncController) clusterOperations(selectedClusters, unselecte
 			clusterObj := clusterObj.(pkgruntime.Object)
 			desiredObj = s.adapter.ObjectForUpdateOp(desiredObj, clusterObj)
 
-			resourceVersion, ok := clusterVersions[clusterName]
+			version, ok := clusterVersions[clusterName]
 			if !ok {
 				// No target version recorded for template+override version
 				operationType = util.OperationTypeUpdate
 			} else {
-				targetVersion := s.adapter.Target().ObjectMeta(clusterObj).ResourceVersion
-				// Versions don't match
-				if resourceVersion != targetVersion {
+				targetVersion := util.GetVersion(s.versionCompareType,
+					s.adapter.Target().ObjectMeta(clusterObj))
+
+				// Check if versions don't match. If they match and the resource uses
+				// Generation to compare versions, then check its ObjectMeta.
+				// This is because Generation is only updated when Spec
+				// changes.
+				if version != targetVersion {
 					operationType = util.OperationTypeUpdate
+				} else if s.versionCompareType == util.Generation {
```

Comment:
Consider skipping the check of the compare type since the equivalent method alone will produce the correct result.